### PR TITLE
[codex] Add granular MCP memory entry validation

### PR DESCRIPTION
**Pull request by Z-M-Huang**
## Summary

- Adds granular support-memory handling for skill-pack artifact import and export flows.
- Extends skill-pack service and registry schemas to validate nested support claims and fragments.
- Adds workflow and registry test coverage plus a short architecture doc update for the support path.

## Impact

This gives reviewers a focused `dev` to `main` PR for the granular MCP memory entry validation work. The compare contains one new commit: `fbb9fa0`.

## Branch notes

`dev` is one commit ahead of `main` and one merge commit behind because `main` contains the prior PR #6 merge commit. I did not merge or rebase `dev`.

## Validation

- `git diff --check origin/main...origin/dev`
- `bash scripts/ci-check.sh`

### CODE REVIEW

**Review by Z-M-Huang — COMMENTED** [review]
## GitVibe Review Matrix

**Status:** `completed`
**Next state:** `changes-required`

Review completed against the GitHub PR head diff for markhuangai/dense-mem#8. One blocking correctness/security issue remains: imported support fragments with an explicit or exported zero source_quality are silently upgraded to trusted/review defaults, which can change support-gate behavior. Role synthesis note: 2 configured member results were available and 1 member result was missing/failed; I used the GitHub PR API because the local checkout did not contain the PR head commit.

### Next Action
Continue with `changes-required`.

### Result
When present, required fixes are posted as pull request review comments. Full structured output remains in the workflow run summary.
Workflow run: https://github.com/markhuangai/dense-mem/actions/runs/26815502544

**Inline comments:** 1

### Required Fixes
1. internal/service/skillpackservice/support.go:269-273 treats any exported support fragment source_quality <= 0 as absent and substitutes mode defaults via sourceQuality(mode). Because internal/service/skillpackservice/types.go:106 allows/omits zero and validateSupportFragment accepts values in [0,1], an artifact containing source_quality 0 is imported as 0.8 or 1.0, which can incorrectly satisfy source-quality promotion gates.

File: internal/service/skillpackservice/support.go [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+		"content_hash": fragmentRes.Fragment.ContentHash,
+		"import_id":    importID,
+	}); err != nil {
+		return "", s.cleanupCreatedEntity(ctx, profileID, "fragment", localID, err)
+	}
+	return localID, nil
+}
+
+func importSourceQuality(mode string, exported float64) float64 {
+	if exported > 0 {
```
Comment (Z-M-Huang) [inline internal/service/skillpackservice/support.go]: This treats a valid exported `source_quality: 0` as absent and upgrades it to the mode default (`0.8` or `1.0`). Source quality feeds claim/fact support gates, so low-quality imported evidence can become stronger than the artifact declares. Preserve explicit zero quality and add a regression test for that case.

Comment (Z-M-Huang) [inline internal/service/skillpackservice/support.go]: This treats explicit `source_quality: 0` as absent and upgrades it to the mode default (`0.8` or `1.0`). Since support fragment validation accepts zero as a valid value and the type uses `omitempty`, imported or round-tripped low-quality evidence can become strong enough to satisfy source-quality promotion gates. Preserve field presence, for example with `*float64` or an equivalent marker, and add a regression test for explicit zero versus omitted quality.

**Review by Z-M-Huang — COMMENTED** [review]
## GitVibe Review Matrix

**Status:** `completed`
**Next state:** `changes-required`

Synthesized 3/3 review-matrix role results for markhuangai/dense-mem#8 and rechecked the PR diff through the GitHub API. One blocking correctness/security issue remains: support fragment source_quality can be explicitly or round-trip exported as zero, but importSourceQuality treats zero as absent and upgrades it to review/trusted defaults, which can change promotion-gate outcomes.

### Next Action
Continue with `changes-required`.

### Result
When present, required fixes are posted as pull request review comments. Full structured output remains in the workflow run summary.
Workflow run: https://github.com/markhuangai/dense-mem/actions/runs/26843424402

**Inline comments:** 1

### Required Fixes
1. internal/service/skillpackservice/support.go:269-273 and internal/service/skillpackservice/types.go:102-106: `SkillPackSupportFragment.SourceQuality` is a non-pointer `float64` with `omitempty`, while `importSourceQuality` returns the mode default whenever the exported value is not greater than zero. Since `validateSupportFragment` accepts values in [0,1], an artifact or export round trip containing explicit `source_quality: 0` is imported as 0.8 or 1.0, which can incorrectly satisfy source-quality promotion gates.

**Review by Z-M-Huang — COMMENTED** [review]
## GitVibe Review Matrix

**Status:** `completed`
**Next state:** `review-passed`

Synthesized 3/3 review-matrix role results for markhuangai/dense-mem#8 and rechecked the current GitHub PR head 756e9c83c53eb85df24c1802ba645d4e6f4679a3. The earlier source_quality blocker is resolved in the current diff by using *float64 presence tracking, preserving explicit zero values during import, and adding a regression test; no blocking correctness, security, regression, or maintainability findings remain.

### Next Action
Continue with `review-passed`.

### Result
When present, required fixes are posted as pull request review comments. Full structured output remains in the workflow run summary.
Workflow run: https://github.com/markhuangai/dense-mem/actions/runs/26850245461

**Review by Z-M-Huang — COMMENTED** [review]
## GitVibe Review Matrix

**Status:** `completed`
**Next state:** `review-passed`

Synthesized 3/3 review-matrix role results for markhuangai/dense-mem#8 and rechecked the current PR head 85f81f937f8f1f56180598e6556e0e57dae98675. The earlier source_quality blocker remains resolved, the one newer compose/startup commit was inspected, and the current CI run passed; no blocking correctness, security, regression, or maintainability findings remain.

### Next Action
Continue with `review-passed`.

### Result
When present, required fixes are posted as pull request review comments. Full structured output remains in the workflow run summary.
Workflow run: https://github.com/markhuangai/dense-mem/actions/runs/26858607228
